src/schedule/edf_schedule.c

```diff
@@ -131,9 +172,15 @@ static int schedule_edf_task(void *data, struct task *task, uint64_t start,
 	return 0;
 }
 
-int schedule_task_init_edf(struct task *task, uint32_t uid,
-			   const struct task_ops *ops,
-			   void *data, uint16_t core, uint32_t flags)
+#if !CONFIG_EDF_SCHEDULER_BUDGETING
+static /* EVen when budgeting is disabled, most of logic is shared with
```

Comment:
Should we just declare this as always public form simplicity ?

Reply:
It shouldn't be reachable when budgeting is not supported. Anyway this if-guarded static keyword is super-ugly and I'll remove it and do it differently.